### PR TITLE
Clarify transient sessions documentation

### LINKED ISSUE #23044: Docs: server_admin/topics/sessions/transient.adoc
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

authentication

### Describe the bug

File: server_admin/topics/sessions/transient.adoc

The documentation describes what is a transient session, but it does not tell how to enable transient sessions in Keycloak for a given client.

### Version

22.0.1

### Expected behavior

Documentation should tell that it can only be obtained for clients with service-accounts enabled, by selecting "use refresh token: off" in the advanced openid connect options of the client.

It should also tell that transient session usage when getting a token for arbitrary users is (unfortunately) NOT possible.

### Actual behavior

Documentation is incomplete.

### How to Reproduce?

Read the doc and try to figure out how to enable transient sessions.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by mposolda**
Closes #23044

### CODE REVIEW

**Comment by mposolda** [comment]
@andymunro Could you please review when you have some free cycles?

File: docs/documentation/server_admin/topics/sessions/transient.adoc [hunk 1]
```diff
@@ -5,5 +5,9 @@
 
 You can conduct transient sessions in {project_name}. When using transient sessions, {project_name} does not create a user session after successful authentication. {project_name} creates a temporary, transient session for the scope of the current request that successfully authenticates the user. {project_name} can run <<_protocol-mappers, protocol mappers>> using transient sessions after authentication.
 
-During transient sessions, the client application cannot refresh tokens, introspect tokens, or validate a specific session. Sometimes these actions are unnecessary, so you can avoid the additional resource use of persisting user sessions. This session saves performance, memory, and network communication (in cluster and cross-data center environments) resources.
+The `sid` and `session_state` of the tokens are usually empty when token is issued with transient sessions. So during transient sessions, the client application cannot refresh tokens or validate a specific session.
```
Comment (andymunro) [inline docs/documentation/server_admin/topics/sessions/transient.adoc:8]:
```suggestion
The `sid` and `session_state` of the tokens are usually empty when the token is issued with transient sessions. So during transient sessions, the client application cannot refresh tokens or validate a specific session.
```

File: docs/documentation/server_admin/topics/sessions/transient.adoc [hunk 2]
```diff
@@ -5,5 +5,9 @@
...
+Sometimes these actions are unnecessary, so you can avoid the additional resource use of persisting user sessions. This session saves performance, memory, and network communication (in cluster and cross-data center environments) resources.
+
+At this moment, transient sessions are automatically used just during <<_service_accounts, service account authentication>> with disabled token refresh. Please note that token refresh is
```
Comment (andymunro) [inline docs/documentation/server_admin/topics/sessions/transient.adoc:11]:
```suggestion
At this moment, transient sessions are automatically used just during <<_service_accounts, service account authentication>> with disabled token refresh. Note that token refresh is
```

**Review by andymunro — CHANGES_REQUESTED** [review]
Thanks, @mposolda. A couple of suggestions.

**Comment by mposolda** [comment]
@andymunro Applied your suggestions. Thanks!

**Pull request merged**